app/controllers/library_hours_controller.rb

```diff
@@ -1,10 +1,43 @@
 # frozen_string_literal: true
 
 class LibraryHoursController < ApplicationController
-  before_action :get_locations, :set_dates, only: [:index]
+  before_action :buildings, :set_dates, only: [:index, :show]
 
   def index
+    @buildings.each do |building|
+      building[:spaces].map! do |space|
+        space = [building[:slug], LibraryHour.where(location_id: space, date: @monday..@sunday + 1)]
+      end
+    end
+  end
+
+  def set_dates
```

Comment:
This logic looks like a setter. I think it should be in `library_hours` model.

Reply:
Did some further review. I see these are controller instance variables, so it's okay.  I'll resolve this conversation.